### PR TITLE
Add descendants' ids to download list when a topic node is selected

### PR DESCRIPTION
**Pull request by lyw07**
# Details

### Summary

* description of the change

This PR is to fix https://github.com/learningequality/kolibri/issues/2692.
Previously, when a topic node is selected, only the topic node would be passed into the download file list, so that none of the resources under this topic node were downloaded.
Currently, if a topic node is selected, then all of its descendants will be downloaded. On the other hand, if a topic node is de-selected, then all of its descendants will be excluded from the download list.

### Reviewer guidance

I'm not sure if the number of queries here is the most optimized. Any suggestions appreciated!

To test the changes:
1. Import content with a topic node selected, then all of its descendants will be imported
2. Import content with a topic node selected and a resource node under that topic node not selected, then all of the topic node's descendants except the one resource node will be imported

### References

* references to related issues and PRs
Issue: https://github.com/learningequality/kolibri/issues/2692

# Contributor Checklist

- [X] PR has the correct target milestone
- [X] PR has the appropriate labels
~- [X] Changes in the PR do not introduce accessibility regressions ([confimed by testing with one of the recommended tools](http://kolibri.readthedocs.io/en/develop/dev/manual_testing.html#accessibility-a11y-testing))~ No accessibility regressions introduced 
- [X] If PR is ready for review, it has been assigned or requests review from someone
~- [X] Documentation is updated as necessary~ No need to update documentation
~- [X] External dependency files are updated (`yarn` and `pip`)~ No external dependency updated
~- [X] If internal dependency is updated, link to diff is included~ No internal dependency updated
~- [X] Screenshots of any front-end changes are in the PR description~ No front-end changes
~- [X] CHANGELOG.rst is updated for high-level changes~ No high-level changes
~- [X] You've added yourself to AUTHORS.rst if you're not there~ My name is there

# Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] PR has been fully tested manually

### CODE REVIEW

**Comment by jonboiser** [comment]
This looks like it will do the job, but is there a way to reduce the code duplication?

File: kolibri/content/management/commands/exportcontent.py [hunk 1]
```diff
@@ -51,6 +51,13 @@ def add_arguments(self, parser):
         parser.add_argument("channel_id", type=str)
         parser.add_argument("destination", type=str)
 
+    def _get_descendants_ids(node_ids):
```
Comment (aronasorman) [inline kolibri/content/management/commands/exportcontent.py]: Maybe `_get_leaves_ids` is a better name.

File: kolibri/content/management/commands/importcontent.py [hunk 2]
```diff
@@ -103,15 +103,24 @@ def download_content(self, channel_id, node_ids=None, exclude_node_ids=None, bas
     def copy_content(self, channel_id, path, node_ids=None, exclude_node_ids=None):
         self._transfer(COPY_METHOD, channel_id, path=path, node_ids=node_ids, exclude_node_ids=exclude_node_ids)
 
+    def _get_descendants_ids(node_ids):
```
Comment (aronasorman) [inline kolibri/content/management/commands/importcontent.py]: Same, change the name here.

**Comment by lyw07** [comment]
@jonboiser I think one way to reduce the duplication might be to move the _get_leaves_ids function to a file under `/content/utils/` folder. But this requires creating an extra file, since I feel like the function doesn't fit into the exiting files under `content/utils`. Do you have any suggestions? Thank you!

**Comment by indirectlylit** [comment]
food for thought: https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)

**Comment by jonboiser** [comment]
@lyw07 It's a judgment call, but I think `_get_leaves_ids` is a sufficiently complex piece of code to warrant implementing in only one place. The idea underlying the rule of three is that you need at least three use cases to make sure you're not overgeneralizing the code you abstract. I'm not sure what the third use case would be, though, since it's import + export.

**Comment by lyw07** [comment]
@indirectlylit @jonboiser Thank you for the suggestions! I will move the code to a new file under `content/utils/` to clean up the duplicated parts :)

**Pull request merged**